quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs

```diff
@@ -543,18 +781,57 @@ fn build_mapping_tree_from_entries<'a>(
     entries: &'a [FieldMappingEntry],
     field_path: &mut Vec<&'a str>,
     schema: &mut SchemaBuilder,
-) -> anyhow::Result<MappingNode> {
+) -> anyhow::Result<(MappingNode, Vec<Field>)> {
```

Comment:
the function name nor the type of the result suggests what this `Vec<Field>` is about.
You need to refactor, rename,  or at least comment.